### ISSUE TITLE
add accessibility tree to PDF export file (Tagged PDF))

### ISSUE THREAD

**Issue by Erwan-le-Gall** [suitenumerique/docs#1132]
label: accessibility

When I use structure (headers, lists…) in a Docs document

Then it should be reflected in the corresponding PDF exported file (= Tagged PDF).

reference: https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf#page=581

This an accessibility requirement.

**Comment by Erwan-le-Gall** [comment]
ping @cyberbaloo

**Comment by Erwan-le-Gall** [comment]
cf https://pdfkit.org/docs/accessibility.html

**Comment by Ovgodd** [comment]
Hello ! 

After investigation, I confirmed that our exported PDFs are not tagged (“Tagged PDF: No” in Acrobat).
This means the document’s logical structure (headings, lists, tables, etc.) is not preserved in the PDF, which fails the accessibility requirement.

The root cause is that @react-pdf/renderer does not currently support generating tagged PDFs.
There’s already an open issue on their repository about this limitation: https://github.com/diegomura/react-pdf/issues/3179.

To meet the requirement, we would need to switch to a PDF generation tool that supports tagged PDF output, such as an HTML→PDF engine (PrinceXML, Antenna House, etc.) or a PDF library with full tagging support.

I tag @virgile-dev

**Comment by virgile-dev** [comment]
Thanks that’s something that will be of interest for @YousefED as well

**Comment by Erwan-le-Gall** [comment]
explicit #758

**Comment by cyberbaloo** [comment]
Maybe look at the side of [Typst](https://typst.app/docs/reference/pdf/#pdf-ua)

> Note that there are some rules in PDF/UA that are crucial for accessibility but cannot be automatically checked. Hence, when exporting a PDF/UA-1 document, make sure you did the following:
> 
> If your document is written in a different language than English, make sure [set the text language](https://typst.app/docs/reference/text/text/#parameters-lang) before any content.
> Make sure to use Typst's semantic elements (like [headings](https://typst.app/docs/reference/model/heading/), [figures](https://typst.app/docs/reference/model/figure/), and [lists](https://typst.app/docs/reference/model/list/)) when appropriate instead of defining custom constructs. This lets Typst know (and export) the role a construct plays in the document. See [the Accessibility guide](https://typst.app/docs/guides/accessibility/) for more details.
> Do not exclusively use contrast, color, format, or layout to communicate an idea. Use text or alternative descriptions instead of or in addition to these elements.
> Wrap all decorative elements without a semantic meaning in [pdf.artifact](https://typst.app/docs/reference/pdf/artifact/).
> Do not use images of text. Instead, insert the text directly into your markup.

**Comment by virgile-dev** [comment]
@Ovgodd I'm closing this here as it needs to be fixed on the libraries side. Please link the issues here when you've created them  on the other repos. Thanks

**Comment by Ovgodd** [comment]
> [@Ovgodd](https://github.com/Ovgodd) I'm closing this here as it needs to be fixed on the libraries side. Please link the issues here when you've created them on the other repos. Thanks

Hello ! this is the issue created on BlockNote side : [2178](https://github.com/TypeCellOS/BlockNote/issues/2178)
